### PR TITLE
[VL][Delta]add more UT for delta lake and enable in CI

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Add time travel and partition filter UT for delta and enable in CI

## How was this patch tested?

UT

### CODE REVIEW

File: gluten-delta/src/main/scala/io/glutenproject/execution/DeltaScanTransformer.scala [hunk 1]
```diff
@@ -14,9 +14,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.glutenproject.extension
+package io.glutenproject.execution
```
Comment (zhztheplayer) [inline gluten-delta/src/main/scala/io/glutenproject/execution/DeltaScanTransformer.scala:17]: So `extension` was by mistake?

**Comment by zhztheplayer** [comment]
Thanks @yma11. And by the way, we may need to add the newly added module folders to [here](https://github.com/oap-project/gluten/blob/0f5d9e18615fc2978bfa286f5dada8e54e0f1acc/.github/workflows/velox_be.yml#L21-L36) and [here](https://github.com/oap-project/gluten/blob/0f5d9e18615fc2978bfa286f5dada8e54e0f1acc/.github/workflows/clickhouse_be_trigger.yml#L22-L31). So that a PR only changes the new modules would trigger CI.

File: gluten-delta/src/main/scala/io/glutenproject/execution/DeltaScanTransformer.scala (hunk 1 above)
Comment (yma11) [reply gluten-delta/src/main/scala/io/glutenproject/execution/DeltaScanTransformer.scala:17]: I think so. I synced with @YannByron and he will take a further look as there may still be some other issues related with column mapping.

**Comment by yma11** [comment]
@zhztheplayer I updated this PR. please help review it again.

**Review by zhztheplayer — APPROVED** [review]
Thanks!

File: .github/workflows/velox_be.yml [hunk 2]
```diff
@@ -24,6 +24,7 @@ on:
       - 'gluten-celeborn/**'
       - 'gluten-core/**'
       - 'gluten-data/**'
+      - 'gluten-delta/**'
```
Comment (zhztheplayer) [inline .github/workflows/velox_be.yml:27]: But should we add `gluten-iceberg/**` too?

**Review by zhztheplayer — APPROVED** [review]

**Comment by yma11** [comment]
Run Gluten Clickhouse CI

**Pull request merged**